implementation-ios.md

```diff
@@ -0,0 +1,94 @@
+
+ To establish a MPTCP connection, you simply have to tweak a bit the configuration used to create the `(NS)URLSession` object. 
+
+1. Create an `(NS)URLSessionConfiguration` object, typically by using the `default` configuration. 
+2. Select the type of service you want. Available types are:
+
+    | Type of service       | Description         | 
+    |:-------------|:------------------|
+    | None     |The default service type indicating that Multipath TCP should not be used.|
+    | Handover |A Multipath TCP service that provides seamless handover between Wi-Fi and cellular in order to preserve the connection.|
+    | Interactive |A service whereby Multipath TCP attempts to use the lowest-latency interface.|
+    | Aggregate   |A service that aggregates the capacities of other Multipath options in an attempt to increase throughput and minimize latency.|
+
+    {: .note }
+    Source for available types of service: [here](https://developer.apple.com/documentation/foundation/nsurlsessionmultipathservicetype) for `NSURLSession` and [here](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/2875967-multipathservicetype) for `URLSession`
+
+3. Set the `multipathServiceType` property of this configuration to the selected type of service. The value is simply `NSURLSessionMultipathServiceType${MODE_NAME}` for `NSURLSession`, and `.{MODE_NAME_LOWER}` for `URLSession`. For example, if we want to use the handover mode, we should pass `NSURLSessionMultipathServiceTypeHandover` to `NSURLSession`, and `.handover` to `URLSession`
+
+4. Create the `(NS)URLSession` by passing in the configuration to the constructor. From now on, every request done with this session should use MPTCP, if the server supports it.
+   
+{: .info }
+To enable the aggregate type of service, you'll probably also need to add the [MPTCP entitlement](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/improving_network_reliability_using_multipath_tcp) to your app. For the other modes, they seem to work fine, even without this additional step, but the aggregate one crashes if you don't set it.
+
+{: .warning }
+The `(NS)URLSession` API is also available on macOS, but you won't be able to set the `multipathServiceType` in the configuration. There is thus no possible way to enable MPTCP with `(NS)URLSession` on macOS yet (though other possibilities exist, see [the FAQ](/faq.html) for more details)
+
+## Examples in different languages
+
+For more detailed examples, please see this [Git
+repository](https://github.com/mptcp-apps/mptcp-hello/). Do not hesitate to
+contribute here and there!
+
+<details markdown="block">
+<summary>Example in Objective-C </summary>
```

Comment:
can you put the example in Swift first? (I understood it was the recommended language now)